File: javascript/bob/bob.js
function asking(input) {
	return input.slice(-1) === "?";
};

function telling(input) {
	return input.trim().length > 0;
};

function yelling(input) {
	return input.toUpperCase() === input && input.toLowerCase() !== input;
};

function Bob() {
	this.hey = function(input) {
		if (yelling(input)) return "Whoa, chill out!";
		else if (asking(input)) return "Sure.";
		else if (telling(input)) return "Whatever.";
		else return "Fine. Be that way!";
	};
};

module.exports = Bob;

File: csharp/bob/Bob.cs
using System.Text.RegularExpressions;

public class Bob
{
  public static string Hey(string input)
  {
    if (Silent(input)) return "Fine. Be that way!";
    else if (Yelling(input)) return "Whoa, chill out!";
    else if (Asking(input)) return "Sure.";
    else return "Whatever.";
  }

  public static bool Silent(string input)
  {
    return input.Trim() == "";
  }

  public static bool Yelling(string input)
  {
    return input.ToUpper() == input && Regex.IsMatch(input, "[a-zA-Z]+");
  }

  public static bool Asking(string input)
  {
    return input.Trim().EndsWith("?");
  }
}
